File: py/1287.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        x = len(arr) / 4

        freq = {}

        for n in arr:
            cur = freq.get(n, 0) + 1

            if float(cur) > x:
                return n
            freq[n] = cur

File: cpp/1287.cpp
class Solution {
  public:
    int findSpecialInteger(vector<int>& arr) {
      float qtr = arr.size() / 4.0;
      std::map<int,int> freq;
      for (int n : arr ) {
        freq[n]++;
        int cur = freq[n];
        if (float(cur) > qtr)
          return n;
      }
      return -1;
    }
};
